service/src/battery.cpp

```diff
@@ -47,6 +48,14 @@ Battery::Battery(Logger* newLogger, bool loglevelSet, QObject *parent) : QObject
     chargerConnectedFile = new QFile("/sys/class/power_supply/usb/present", this);
     logE("Charger status file: " + chargerConnectedFile->fileName());
 
+    // Number: temperature
+    temperatureFile   = new QFile("/sys/class/power_supply/battery/temp", this);
+    logE("Temperature file: " + temperatureFile->fileName());
+
+    // String: Good, Warm, Overheat, (others?)
+    healthFile   = new QFile("/sys/class/power_supply/battery/health", this);
```

Comment:
I'm interested in what other values there may be, but let's get the basics done first.

Reply:
I'm not sure where to get the relevant driver kernel sources, but for 4.9.230 kernel I find these:

 `drivers/power/supply/power_supply_sysfs.c` 

        static const char * const power_supply_health_text[] = {
                "Unknown", "Good", "Overheat", "Dead", "Over voltage",
                "Unspecified failure", "Cold", "Watchdog timer expire",
                "Safety timer expire", "Over current"
        };

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/drivers/power/supply/power_supply_sysfs.c?h=v4.9.230&id=eb5d955823b582625d7f157125ea55e6ac1b60e4

while `include/linux/power_supply.h` has:

        enum {
                POWER_SUPPLY_HEALTH_UNKNOWN = 0,
                POWER_SUPPLY_HEALTH_GOOD,
                POWER_SUPPLY_HEALTH_OVERHEAT,
                POWER_SUPPLY_HEALTH_DEAD,
                POWER_SUPPLY_HEALTH_OVERVOLTAGE,
                POWER_SUPPLY_HEALTH_UNSPEC_FAILURE,
                POWER_SUPPLY_HEALTH_COLD,
                POWER_SUPPLY_HEALTH_WATCHDOG_TIMER_EXPIRE,
                POWER_SUPPLY_HEALTH_SAFETY_TIMER_EXPIRE,
                POWER_SUPPLY_HEALTH_OVERCURRENT,
        };

https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/tree/include/linux/power_supply.h?h=v4.9.230&id=eb5d955823b582625d7f157125ea55e6ac1b60e4

of course, qcom doesn't seem to use that, and maps *some* status values to health, other drivers may do different things:

      #define STATUS_USBIN_VALID      BIT(0) /* USB connection is valid */
      #define STATUS_DCIN_VALID       BIT(1) /* DC connection is valid */
      #define STATUS_BAT_HOT          BIT(2) /* Battery temp 1=Hot, 0=Cold */
      #define STATUS_BAT_OK           BIT(3) /* Battery temp OK */
      #define STATUS_BAT_PRESENT      BIT(4) /* Battery is present */
      #define STATUS_CHG_DONE         BIT(5) /* Charge cycle is complete */
      #define STATUS_CHG_TRKL         BIT(6) /* Trickle charging */
      #define STATUS_CHG_FAST         BIT(7) /* Fast charging */
      #define STATUS_CHG_GONE         BIT(8) /* No charger is connected */
      [...]
        case POWER_SUPPLY_PROP_HEALTH:
                if (status & STATUS_BAT_OK)
                        val->intval = POWER_SUPPLY_HEALTH_GOOD;
                else if (status & STATUS_BAT_HOT)
                        val->intval = POWER_SUPPLY_HEALTH_OVERHEAT;
                else
                        val->intval = POWER_SUPPLY_HEALTH_COLD;
                break;